### PR TITLE
Fix `model` parameter in Pose, Segment dataset train examples

### PR DESCRIPTION
**Pull request by lunarifish**
Hi! I am following the [training yolov8-pose models on COCO8-Pose dataset](https://docs.ultralytics.com/datasets/pose/coco8-pose/#usage) guide, it seems like in the "usage" section, the pre-trained model used in shell command differed from the one used in python script. The shell command will start a training session using `yolov8n.pt`, instead of `yolov8n-pose.pt`:

## Python
```python
...
model = YOLO("yolov8n-pose.pt")  # load a pretrained model (recommended for training)
...
```

## CLI
```
yolo detect train data=coco8-pose.yaml model=yolov8n.pt epochs=100 imgsz=640
```

I have found the same situation in some of the other dataset examples, please have a look of the commit. Are these errors?

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://github.com/ultralytics/actions)<sub>

### 🌟 Summary
Updated documentation to correctly reference specific pretrained models for different tasks.

### 📊 Key Changes
- **COCO-Pose Documentation**: Changed model reference from `yolov8n.pt` to `yolov8n-pose.pt`.
- **COCO8-Pose Documentation**: Changed model reference from `yolov8n.pt` to `yolov8n-pose.pt`.
- **Tiger-Pose Documentation**: Changed model reference from `yolov8n.pt` to `yolov8n-pose.pt`.
- **COCO-Seg Documentation**: Changed model reference from `yolov8n.pt` to `yolov8n-seg.pt`.
- **COCO8-Seg Documentation**: Changed model reference from `yolov8n.pt` to `yolov8n-seg.pt`.

### 🎯 Purpose & Impact
- **Clarity**: Ensures that users are training with the correct YOLOv8 models specific to pose and segmentation tasks, improving their experience and results. 🚀
- **Accuracy**: Reduces confusion and potential errors in model training by clearly specifying the appropriate pretrained models. 👍

### CODE REVIEW

**Comment by lunarifish** [comment]
I have read the CLA Document and I sign the CLA

**Comment by glenn-jocher** [comment]
@lunarifish ah thank you, yes these updates are great, the current content is incorrect.
